BenadrylAutomation/src/main/java/com/jnj/pageobjects/WhereToBuyPage.java

```diff
@@ -0,0 +1,203 @@
+/**
+ * @author Goutam Naik
+ * @date 27-March-24
+ */
+
+package com.jnj.pageobjects;
+
+import java.util.List;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.testng.Assert;
+
+import com.jnj.actions.Action;
+import com.jnj.base.BaseClass;
+
+public class WhereToBuyPage extends BaseClass {
+	String actualText;
+
+	/**
+	 * Constructor
+	 */
+	public WhereToBuyPage() {
+		PageFactory.initElements(driver, this);
+	}
+
+	/**
+	 * Locators
+	 */
+	@FindBy(css = "#content-main")
+	private WebElement headingText;
+
+	@FindBy(css = "input[placeholder='Find a specific product']")
+	private WebElement specificProductSearchBox;
+
+	@FindBy(css = "span[class='spinner']")
+	private WebElement spinner;
+
+	@FindBy(xpath = "//ul[contains(@id,'product-finder-result-listbox')]")
+	private WebElement prodResultListbox;
+
+	@FindBy(xpath = "//div[contains(@class,'ps-product-details-title')]/label")
+	private WebElement resProductName;
+
+	@FindBy(css = "#__ps-sku-selector-0_0")
+	private WebElement categoryDropdown;
+
+	@FindBy(css = "#__ps-sku-selector-2_0")
+	private WebElement sizeDropdown;
+
+	@FindBy(css = "#__ps-map-location-textbox_0")
+	private WebElement locationInput;
+
+	@FindBy(xpath = "//div[@class='ps-address']/div[1]/div[1]")
+	private List<WebElement> addresses;
+
+	private WebElement getOption(String option) {
+		return driver.findElement(By.xpath("//li[@role='option']/span[contains(text(),'" + option + "')]"));
+	}
+
+	private WebElement getStoreViewbtn(String storeName) {
+		return driver.findElement(By.xpath("//button[contains(@aria-label,'" + storeName + "')]"));
+	}
+
+	/**
+	 * Function to verifyheading on the Where to buy page
+	 * 
+	 * @param expectedHeader pass expected header
+	 */
+	public void verifyHeading(String expectedHeader) {
+		actualText = headingText.getText().trim();
+		expectedHeader = expectedHeader.trim();
+		Action.printAndAssert(actualText, expectedHeader);
+		BaseClass.extentInfoLog("Succesfully verifed heading : " + actualText);
+	}
+
+	/**
+	 * Function to initiate product search
+	 * @param productName pass product name
+	 */
+	public void initiateProductSearch(String productName) {
+		Action.explicitWait(specificProductSearchBox, 10);
+		Action.performActionwithExtentInfoLog(headingText, "sendKeys", "Sending keys to product search", productName);
+	}
+
+	/**
+	 * Function to search a product
+	 * 
+	 * @param productKey pass product name to search
+	 */
+	public void searchProduct(String productKey) {
+		Action.explicitWait(specificProductSearchBox, 20);
+		Action.explicitWaitForElementToDisappear(spinner, 20);
+		Action.performActionwithExtentInfoLog(specificProductSearchBox, "click", "Clicked inside product searchbox");
+		Action.waitFor(2000);
```

Comment:
@goutamnaik007 this is needed?

Reply:
yes. If i remove this, even if i enter the search criteria, the result listbox is not displayed.
@vaibhavNagvekarSJI

Reply:
fixed